src/shared/MovieCard.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div class="movie-card" @click="onDetail">
+    <div class="movie-card__header">
+      <img class="movie-card__img" :src="movie?.poster_path" height="500" />
+    </div>
+    <div class="movie-card__body">
+      <div class="movie-card__line">
+        <div class="movie-card__title">
+          {{ movie?.title }}
+        </div>
+        <span class="movie-card__release_date">
+          {{ movie?.release_date }}
+        </span>
+      </div>
+      <p class="movie-card__genres">{{ movie?.genres.join(" & ") }}</p>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+import { IMovie } from "../interfaces/movie.interface";
+
+export default defineComponent({
+  name: "MovieCard",
+  props: {
+    movie: Object as () => IMovie,
```

Comment:
as I see you pass onDetail callback for this component in the MovieList, and I agree with this approach, maybe not for all cards on click should handle redirect, so add onDetail to the props and remove it from methods